compiler/rustc_expand/src/mbe/macro_rules.rs

```diff
@@ -205,6 +207,29 @@ fn trace_macros_note(cx_expansions: &mut FxIndexMap<Span, Vec<String>>, sp: Span
     cx_expansions.entry(sp).or_default().push(message);
 }
 
+pub(super) trait Tracker<'matcher> {
+    /// This is called before trying to match next MatcherLoc on the current token
```

Comment:
Please add a full stop at the end of the comment, and all the other ones in this trait.